File: 0092 - Reverse Linked List II/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseBetween(struct ListNode* head, int left, int right){
    struct ListNode dummy, *pre = &dummy, *tail, *tmp;
    dummy.next = head;
    for(int i = 0; i < left-1; i++) pre = pre->next;
    tail = pre->next;
    for(int i = 0; i < right-left; i++){
        tmp = pre->next;
        pre->next = tail->next;
        tail->next = pre->next->next;
        pre->next->next = tmp;
    }
    return dummy.next;
}

File: 0092 - Reverse Linked List II/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        self.next, pre = head, self
        for _ in range(left-1):
            pre = pre.next
        tail = pre.next
        for _ in range(right-left):
            tmp = pre.next
            pre.next = tail.next
            tail.next = pre.next.next
            pre.next.next = tmp
        return self.next